precompiles/crowdloan-rewards/src/lib.rs

```diff
@@ -0,0 +1,274 @@
+
+		log::trace!(target: "crowdloan-rewards-precompile", "Gonna call with origin {:?}", origin);
+
+		match outer_call.dispatch(Some(origin).into()) {
+			Ok(post_info) => {
+				let gas_used = Runtime::GasWeightMapping::weight_to_gas(
+					post_info.actual_weight.unwrap_or(info.weight),
+				);
+				Ok(PrecompileOutput {
+					exit_status: ExitSucceed::Stopped,
+					cost: gas_used,
+					output: Default::default(),
+					logs: Default::default(),
+				})
+			}
+			Err(e) => {
+				log::trace!(
+					target: "crowdloan-rewards-precompile",
+					"Crowdloan rewards call via evm failed {:?}",
+					e
+				);
+				Err(ExitError::Other(
+					"Crowdloan rewards call via EVM failed".into(),
+				))
+			}
+		}
+	}
+}
+
+impl<Runtime> CrowdloanRewardsWrapper<Runtime>
+where
+	Runtime: pallet_crowdloan_rewards::Config + pallet_evm::Config + frame_system::Config,
+	Runtime::AccountId: From<H160>,
+	BalanceOf<Runtime>: TryFrom<U256> + TryInto<u128> + Debug,
+	Runtime::Call: Dispatchable<PostInfo = PostDispatchInfo> + GetDispatchInfo,
+	<Runtime::Call as Dispatchable>::Origin: From<Option<Runtime::AccountId>>,
+	Runtime::Call: From<pallet_crowdloan_rewards::Call<Runtime>>,
+{
+	// The accessors are first. They directly return their result.
+	fn is_contributor(
+		input: &[u8],
+		target_gas: Option<u64>,
+	) -> Result<PrecompileOutput, ExitError> {
+		// parse the address
+		let contributor = H160::from_slice(&input[12..32]);
+
+		log::trace!(
+			target: "crowdloan-rewards-precompile",
+			"Checking whether {:?} is a contributor",
+			contributor
+		);
+
+		let gas_consumed = <Runtime as pallet_evm::Config>::GasWeightMapping::weight_to_gas(
+			<Runtime as frame_system::Config>::DbWeight::get().read,
+		);
+
+		// Make sure enough gas
+		if let Some(gas_limit) = target_gas {
+			if gas_consumed > gas_limit {
+				return Err(ExitError::OutOfGas);
+			}
+		}
+
+		let account: Runtime::AccountId = contributor.into();
+		// fetch data from pallet
+		let is_contributor =
+			pallet_crowdloan_rewards::Pallet::<Runtime>::accounts_payable(account).is_some();
+
+		log::trace!(target: "crowldoan-rewards-precompile", "Result from pallet is {:?}", is_contributor);
+
+		let gas_consumed = <Runtime as pallet_evm::Config>::GasWeightMapping::weight_to_gas(
+			<Runtime as frame_system::Config>::DbWeight::get().read,
+		);
+
+		Ok(PrecompileOutput {
+			exit_status: ExitSucceed::Returned,
+			cost: gas_consumed,
+			output: bool_to_solidity_bytes(is_contributor),
+			logs: Default::default(),
+		})
+	}
+
+	fn reward_info(input: &[u8], target_gas: Option<u64>) -> Result<PrecompileOutput, ExitError> {
+		// parse the address
+		let contributor = H160::from_slice(&input[12..32]);
+
+		log::trace!(
+			target: "crowdloan-rewards-precompile",
+			"Checking reward info for {:?}",
+			contributor
+		);
+
+		let account: Runtime::AccountId = contributor.into();
+
+		let gas_consumed = <Runtime as pallet_evm::Config>::GasWeightMapping::weight_to_gas(
```

Comment:
Why is this one read instead of using something like `accounts_payable()`'s weight?

Reply:
Account Payable is a storage item not an extrinsic, so it does not have per se a weight associated from benchmarking. It should coincide with one DB read though

Reply:
Ah, good point.